CMakeLists.txt

```diff
@@ -1,7 +1,10 @@
 ########################################################################
 # Project setup
 ########################################################################
-cmake_minimum_required(VERSION 2.8.9)
+
+cmake_minimum_required(VERSION 3.20)
+add_compile_definitions(QT_DISABLE_DEPRECATED_BEFORE=0x050F00)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
@guruofquality With Pothos globally being C++11 and requiring a lower version of CMake, this appears to have larger consequences. Ten years out from C++11, perhaps we should update in a larger version bump and tie this PothosFlow change to that.

@sielicki With this extensive a CMake version requirement, what features are required from this version? I'm not seeing anything obvious, so do the Qt scripts require it?

Reply:
I'm not sure off the top of my head what version of CMake ships with the latest Ubuntu LTS, but I'd say we should be compatible with that.

Reply:
Newer versions of cmake are starting to complain that `compatibility with CMake < 2.8.12 will be removed from a future version of CMake.`, so I just wanted to make it shut up about that. Even 18.04 ships CMake 3.10, so I do think it's safe to bump the minimum up a bit, but 3.20 is way too far.

Reply:
bumping the standard to C++17 is a hard requirement of Qt6

Reply:
From https://doc.qt.io/qt-6/cmake-get-started.html

> cmake_minimum_required() specifies the minimum CMake version that the application requires. [Qt6] itself requires at least CMake version 3.16.

unfortunately.